### PR TITLE
chore: Rename trace-collector to trace-gateway

### PR DESCRIPTION
**Pull request by TeodorSAP**
## Description

Changes proposed in this pull request (what was done and why):

- Rename `trace-collector` resources and instances to `trace-gateway`
- Adjust documentation accordingly
- Implement deletion logic for old `trace-collector` resources

Changes refer to particular issues, PRs or documents:

- Resolves https://github.com/kyma-project/telemetry-manager/issues/652

## Traceability
- [x] The PR is linked to a GitHub issue.
- [ ] The follow-up issues (if any) are linked in the `Related Issues` section.
- [x] If the change is user-facing, the documentation has been adjusted.
- [ ] If a CRD is changed, the corresponding Busola ConfigMap has been adjusted.
- [ ] The feature is unit-tested.
- [x] The feature is e2e-tested.

### CODE REVIEW

**Comment by NHingerl** [comment]
I'll adjust the docs in DITA once we're ready. 
Does this have to happen together with the 1.25 release?

**Comment by a-thaler** [comment]
yes, that changes are tightly related to 1.25.0

File: internal/reconciler/tracepipeline/reconciler.go [hunk 1]
```diff
@@ -286,3 +297,72 @@ func (r *Reconciler) getReplicaCountFromTelemetry(ctx context.Context) int32 {
...
+					Namespace: "kyma-system",
+				},
+			})
+	}
+
+	for _, oldResource := range oldTraceCollectorResources {
+		if err := r.Delete(ctx, oldResource); err != nil && !apierrors.IsNotFound(err) {
+			return fmt.Errorf("failed to delete one of the old trace collector resources: %w", err)
+		}
+	}
```
Comment (skhalash) [inline internal/reconciler/tracepipeline/reconciler.go:365]: I suggest modifying the error handling logic to attempt deleting all possible resources first. If any deletion fails, an error can be returned at the end. You can use https://pkg.go.dev/errors#Join for that.

**Review by skhalash — COMMENTED** [review]
Looks very good, just added a small comment

File: internal/reconciler/tracepipeline/reconciler.go (hunk 1 above)
Comment (TeodorSAP) [reply internal/reconciler/tracepipeline/reconciler.go:365]: Alright, thank you!
I will do the changes tomorrow morning

**Comment by NHingerl** [comment]
> yes, that changes are tightly related to 1.25.0

Wait, the milestone label says 1.26, so I thought it's not extremely urgent.
Aaargh. Reviewing now.

**Comment by a-thaler** [comment]
ahh, sorry, no, it is for 1.26. I mixed it up with a different ticket

File: docs/user/03-traces.md [hunk 2]
```diff
@@ -522,4 +522,4 @@ If you just want to see traces for one particular request, you can manually forc
 
 **Cause**: Gateway cannot receive spans at the given rate.
 
-**Remedy**: Manually scale out the gateway by increasing the number of replicas for the Trace collector. See [Module Configuration and Status](https://kyma-project.io/#/telemetry-manager/user/01-manager?id=module-configuration).
+**Remedy**: Manually scale out the gateway by increasing the number of replicas for the Trace gateway. See [Module Configuration and Status](https://kyma-project.io/#/telemetry-manager/user/01-manager?id=module-configuration).
```
Comment (NHingerl) [inline docs/user/03-traces.md:525]:
```suggestion
**Remedy**: Manually scale out the gateway by increasing the number of replicas for the trace gateway. See [Module Configuration and Status](https://kyma-project.io/#/telemetry-manager/user/01-manager?id=module-configuration).
```

For consistency. It's lower case everywhere else.

**Review by NHingerl — COMMENTED** [review]
- line 49 also talks about "the collector". I assume that stays unchanged but please confirm.
- in line 432, the "c" of OTel Collector should be capitalized for consistency with all the other occurrences.

File: internal/reconciler/tracepipeline/reconciler.go [hunk 3]
```diff
@@ -286,3 +297,77 @@ func (r *Reconciler) getReplicaCountFromTelemetry(ctx context.Context) int32 {
...
+		if err := r.Delete(ctx, oldResource); err != nil && !apierrors.IsNotFound(err) {
+			errs = errors.Join(errs, fmt.Errorf("failed to delete old trace collector resource: %w", err))
+		}
+	}
+
+	if errs != nil {
+		return errs
+	}
+
+	return nil
```
Comment (skhalash) [inline internal/reconciler/tracepipeline/reconciler.go:372]:
```suggestion
	return errs
```

File: docs/user/03-traces.md (hunk 2 above)
Comment (TeodorSAP) [reply docs/user/03-traces.md:525]: - line 49 stays unchanged, there we talk about the Otel Collector that gets deployed
- Standardized all occurrences of "Otel collector" to use capitalized "C"

File: docs/contributor/arch/013-resource-attributes-enrichment-for-non-workload-metrics.md [hunk 4]
```diff
@@ -34,7 +34,7 @@ The `Output Pipeline` contains the rest of the processors and the `otlp` exporte
...
+To check the effect of the new setup on the performance of the metric gateway, [load tests](https://github.com/kyma-project/telemetry-manager/tree/main/docs/contributor/benchmarks#metric-gateway) were executed using version [1.22.0](https://github.com/kyma-project/telemetry-manager/releases/tag/1.22.0) for the telemetry-manager and the OTel Collector image `europe-docker.pkg.dev/kyma-project/dev/kyma-otel-collector:PR-121`, which contains the [routingconnector](https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/connector/routingconnector) and [forwardconnector](https://github.com/open-telemetry/opentelemetry-collector/tree/main/connector/forwardconnector) components.
```
Comment (NHingerl) [inline docs/contributor/arch/013-resource-attributes-enrichment-for-non-workload-metrics.md:37]: thanks for spotting these occurrences!!

**Review by NHingerl — DISMISSED** [review]

**Comment by NHingerl** [comment]
> ahh, sorry, no, it is for 1.26. I mixed it up with a different ticket

LOL thanks for the new grey hair I got ;)

**Review by skhalash — APPROVED** [review]

**Review by NHingerl — APPROVED** [review]

**Pull request merged**